nginx/nginx_test.go

```diff
@@ -100,23 +100,15 @@ func TestStopWaitsForGracefulShutdownOfNginx(t *testing.T) {
 	assert.Error(lb.Health(), "should have waited for nginx to gracefully stop")
 }
 
-func TestNginxNotStartedUntilFirstUpdate(t *testing.T) {
-	tmpDir := setupWorkDir(t)
-	lb := newUpdaterWithBinary(tmpDir, "./fake_graceful_nginx.py")
-
-	assert.NoError(t, lb.Start())
-	assert.Nil(t, lb.nginx.Process)
-}
-
 func TestNginxStartedAfterFirstUpdate(t *testing.T) {
 	tmpDir := setupWorkDir(t)
-	lb := newUpdaterWithBinary(tmpDir, "./fake_graceful_nginx.py")
+	lb := newUpdaterWithBinary(tmpDir, "./fake_failing_nginx.sh")
 
-	assert.NoError(t, lb.Start())
-	assert.NoError(t, lb.Update(controller.IngressUpdate{Entries: []controller.IngressEntry{{
+	lb.Start()
+	err := lb.Update(controller.IngressUpdate{Entries: []controller.IngressEntry{{
 		Host: "james.com",
-	}}}))
-	assert.NotNil(t, lb.nginx.Process.Pid)
+	}}})
+	assert.EqualError(t, err, "nginx died shortly after starting")
```

Comment:
This test is duplicating  `TestFailsIfNginxDiesEarly`. Instead we could have it execute a non-existent command, that would cause it to fail immediately.